vaultwarden: update to 1.21.0, rename from bitwarden_rs

K900 opened the pull request:
I tried to make this as non-breaking as possible, but it will still
break things slightly for people expecting certain file names in the
packages themselves.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

K900 commented:
Added a release note, it seems appropriate judging by the other ones.

K900 commented:
I'm not sure why ofborg is failing here. Is it because it can't fetch the source?

K900 commented:
I gave up on making ofborg play nice with Cargo.lock, let's just stick to the old way for now.

aanderse commented on nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:
```diff
@@ -75,7 +75,13 @@
+    <itemizedlist spacing="compact">
+      <listitem>
+        <para>
+          The <literal>bitwarden_rs</literal> packages and modules were
```
Would the present state of the code would cause this service to fail for existing users because of ownership permissions?

K900 replied:
Nope, systemd will automatically change the ownership of `StateDirectory` to the user that owns the service, so we're good here.

aanderse replied:
Ignore this, see above comment.

aanderse reviewed (commented):
I've listed a few things you could do to make this change backwards compatible for existing users, if you were inclined to do so.

K900 commented on nixos/modules/services/security/vaultwarden/default.nix:
```diff
@@ -3,9 +3,9 @@
-  user = config.users.users.bitwarden_rs.name;
-  group = config.users.groups.bitwarden_rs.name;
+  cfg = config.services.vaultwarden;
+  user = config.users.users.vaultwarden.name;
+  group = config.users.groups.vaultwarden.name;
```
Systemd should sort out the permissions on `/var/lib/bitwarden_rs` automatically, so this shouldn't be a problem unless people go out of their way to modify the users in their own config, which I really hope no one does.

aanderse replied:
Gosh you're right! I was mistaken.

So what is the incompatibility this PR introduces?

K900 replied:
People manually modifying the users/units in their configuration will have things explode because the unit is no longer there. In most cases it will explode in a pretty obvious way, so this is probably fine?

K900 replied:
Also anything that depends on the paths in the final derivations, like custom units and such.

K900 commented on nixos/modules/services/security/vaultwarden/default.nix:
```diff
@@ -131,15 +135,15 @@ in {
+    systemd.services.backup-vaultwarden = mkIf (cfg.backupDir != null) {
+      description = "Backup vaultwarden";
       environment = {
         DATA_FOLDER = "/var/lib/bitwarden_rs";
```
That will still require unstable users to migrate manually, right?

aanderse replied:
Ignore this, see above comment.

Mic92 replied:
Right. It could move atomically from the old to the new location so.

K900 commented on nixos/modules/services/security/vaultwarden/default.nix:
```diff
@@ -106,20 +110,20 @@ in {
+    users.groups.vaultwarden = { };
 
-    systemd.services.bitwarden_rs = {
+    systemd.services.vaultwarden = {
```
Thanks, added!

aanderse commented on nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:
```diff
@@ -75,7 +75,13 @@
+    <itemizedlist spacing="compact">
+      <listitem>
+        <para>
+          The <literal>bitwarden_rs</literal> packages and modules were
```
You should list explicitly that the user and service names changed and anything else that that might be relevant. If users are required to take any action you could mention here, if not you could mention that was well.

K900 replied:
I've added a more detailed release note, but I think it might be _too_ detailed now.

aanderse replied:
hahaha probably - but that is much better than too little :+1:

aanderse reviewed (changes requested):
I think the change looks good. Thanks for your work on this.

Could you please just add a little more detail to the release notes and I think we're good to go.

K900 commented:
Rebased on top of latest master, is there anything else stopping this from getting merged?

aanderse reviewed (approved):
LGTM but I would prefer if @msteen, or anyone else familiar with this package & module, to approve if possible before we merge.

If we don't hear anything back in a week or so ping me and we'll go ahead and merge.

K900 committed:
vaultwarden: update to 1.22.1, rename from bitwarden_rs

I tried to make this as non-breaking as possible, but it will still
break things slightly for people expecting certain file names in the
packages themselves.

K900 commented:
Rebased again, and updated to 1.22.1. I squashed all the commits because rebasing individual release notes every time was too obnoxious.
